tests/lib/check/compliance_check_test.py

```diff
@@ -0,0 +1,166 @@
+                ResourceIdTemplate="template1",
+                Severity="high",
+                ResourceType="resource1",
+                Description="Description 1",
+                Risk="risk1",
+                RelatedUrl="url1",
+                Remediation={
+                    "Code": {
+                        "CLI": "cli1",
+                        "NativeIaC": "native1",
+                        "Other": "other1",
+                        "Terraform": "terraform1",
+                    },
+                    "Recommendation": {"Text": "text1", "Url": "url1"},
+                },
+                Categories=["categoryone"],
+                DependsOn=["dependency1"],
+                RelatedTo=["related1"],
+                Notes="notes1",
+                Compliance=[],
+            ),
+            "check2": Check_Metadata_Model(
+                Provider="aws",
+                CheckID="check2",
+                CheckTitle="Check 2",
+                CheckType=["type2"],
+                ServiceName="service2",
+                SubServiceName="subservice2",
+                ResourceIdTemplate="template2",
+                Severity="medium",
+                ResourceType="resource2",
+                Description="Description 2",
+                Risk="risk2",
+                RelatedUrl="url2",
+                Remediation={
+                    "Code": {
+                        "CLI": "cli2",
+                        "NativeIaC": "native2",
+                        "Other": "other2",
+                        "Terraform": "terraform2",
+                    },
+                    "Recommendation": {"Text": "text2", "Url": "url2"},
+                },
+                Categories=["categorytwo"],
+                DependsOn=["dependency2"],
+                RelatedTo=["related2"],
+                Notes="notes2",
+                Compliance=[],
+            ),
+        }
+
+    def test_update_checks_metadata(self):
+        bulk_compliance_frameworks = self.get_custom_framework()
+        bulk_checks_metadata = self.get_custom_check_metadata()
+
+        updated_metadata = update_checks_metadata_with_compliance(
+            bulk_compliance_frameworks, bulk_checks_metadata
+        )
+
+        assert "check1" in updated_metadata
+        assert "check2" in updated_metadata
+        assert "manual_check" in updated_metadata
+
+        assert len(updated_metadata["check1"].Compliance) == 1
+        assert len(updated_metadata["check2"].Compliance) == 1
+        assert len(updated_metadata["manual_check"].Compliance) == 1
+
+        manual_compliance = updated_metadata["manual_check"].Compliance[0]
+        assert manual_compliance.Framework == "Framework1"
+        assert manual_compliance.Provider == "Provider1"
+        assert manual_compliance.Version == "1.0"
+        assert manual_compliance.Description == "Framework 1 Description"
+
+    def test_update_checks_metadata_with_manual_control(self):
```

Comment:
What is the difference between this test and the one above? Could we assert more fields to be sure that everything is right?

Reply:
Yes, one test it's enough for this behaviour